src/test/java/com/marklogic/appdeployer/command/security/DeploySecureCredentialsTest.java

```diff
@@ -0,0 +1,21 @@
+package com.marklogic.appdeployer.command.security;
+
+import com.marklogic.appdeployer.AbstractAppDeployerTest;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+
+public class DeploySecureCredentialsTest extends AbstractAppDeployerTest {
```

Comment:
Because the manager extends `AbstractResourceManager`, change this to extend `AbstractManageResourceTest`. Remove the test method in here and move the sample creds file to src/test/resources/sample-app/src/main/ml-config/security/secure-credentials. And change it to use `rest-reader`  as there's no need for a custom role here. 

You can then implement the methods required by the parent class like this:

```
@Override
protected ResourceManager newResourceManager() {
	return new SecureCredentialsManager(manageClient);
}

@Override
protected Command newCommand() {
	return new DeploySecureCredentialsCommand();
}

@Override
protected String[] getResourceNames() {
	return new String[]{"sec-creds1"};
}
```

The parent class will do a fairly thorough job of ensuring that create and delete both work.